internal/api/v1/handlers/instance.go

```diff
@@ -156,3 +156,127 @@ func (h *InstanceHandler) GetInstance(c *fiber.Ctx) error {
 
 	return c.JSON(instance)
 }
+
+// GetPublicIPs returns a list of public IPs and their associated job IDs
+func (h *InstanceHandler) GetPublicIPs(c *fiber.Ctx) error {
+	fmt.Println("🔍 Getting public IPs...")
+
+	// Get instances with their public IPs using the service
+	instances, err := h.service.GetPublicIPs(c.Context())
+	if err != nil {
+		fmt.Printf("❌ Error getting public IPs: %v\n", err)
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": fmt.Sprintf("failed to get public IPs: %v", err),
+		})
+	}
+
+	fmt.Printf("✅ Found %d instances\n", len(instances))
+
+	// Convert instances to a slice of maps with only IP and job ID
+	ipMaps := make([]map[string]interface{}, len(instances))
+	for i, instance := range instances {
+		ipMaps[i] = map[string]interface{}{
+			"public_ip": instance.PublicIP,
+			"job_id":    instance.JobID,
+		}
+	}
+
+	// Return only the public IPs and job IDs
+	return c.JSON(fiber.Map{
+		"instances": ipMaps,
+		"total":     len(instances),
+	})
+}
+
+// GetAllMetadata returns a list of all instance details
+func (h *InstanceHandler) GetAllMetadata(c *fiber.Ctx) error {
+	fmt.Println("🔍 Getting all instance metadata...")
+
+	// Get instances with their details using the service
+	instances, err := h.service.GetPublicIPs(c.Context())
+	if err != nil {
+		fmt.Printf("❌ Error getting instance metadata: %v\n", err)
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": fmt.Sprintf("failed to get instance metadata: %v", err),
+		})
+	}
+
+	fmt.Printf("✅ Found %d instances\n", len(instances))
+
+	// Convert instances to a slice of maps with all metadata
+	instanceMaps := make([]map[string]interface{}, len(instances))
+	for i, instance := range instances {
+		instanceMaps[i] = map[string]interface{}{
+			"id":         instance.ID,
+			"job_id":     instance.JobID,
+			"name":       instance.Name,
+			"public_ip":  instance.PublicIP,
+			"region":     instance.Region,
+			"size":       instance.Size,
+			"image":      instance.Image,
+			"tags":       instance.Tags,
+			"status":     instance.Status.String(),
+			"created_at": instance.CreatedAt,
+		}
```

Comment:
instead of converting it to some other map. why not using the same type that is defined in the models pkg for db?
have a look at this: https://github.com/celestiaorg/talis/blob/0eb1540e4d52b4c76800a3a5162182a8124c2900/internal/api/v1/handlers/instance.go#L29